app/src/api/multishot.lisp

```diff
@@ -69,12 +81,49 @@
                    (= 3 (length measurement-noise)))
                (every #'realp measurement-noise)))
 
+  (cond
+    ;; If we have too many qubits or too few trials, just do things
+    ;; single-threaded.
+    ((or (not *single-user-mode*)
+         (>= num-qubits qvm::*qubits-required-for-parallelization*)
+         (< num-trials *trials-required-for-parallelization*)
```

Comment:
I find it slightly confusing that `*trials-required-for-parallelization*` is a lower bound for parallelization, while `qvm::*qubits-required-for-parallelization*` is an upper bound.

Hmm, grepping through the QVM code, it looks like `*qubits-required-for-parallelization*` is elsewhere used as a lower bound for parallelization within the VM? I guess the reason it's used as an upper limit here is that more qubits implies greater memory consumption and you don't want to overcommit?

Reply:
They're both of the same spirit. Number of trials before you parallelize, number of qubits before you parallelize.

The thing is, we _do_ parallelize if num-qubits exceeds that number, we just don't parallelize _here_.

Reply:
Should these be independent numbers of qubits at least?

My not-fire-tested guess is that in gate application you'd want to parallelize when you have enough work to bother farming it out and that here you'd want to parallelize when the result tables to have sufficient overlap for the merge process to be a savings; and my second not-fire-tested guess is that those two tipping points don't have much to do with each other.